addons/mrp_byproduct/mrp_byproduct.py

```diff
@@ -123,6 +123,10 @@ def _get_subproduct_factor(self, cr, uid, production_id, move_id=None, context=N
             if subproduct_record.bom_id.product_qty:
                 subproduct_factor = subproduct_record.product_qty / subproduct_record.bom_id.product_qty
                 return subproduct_factor
+        sub_id = sub_obj.search(cr, uid, [('product_id', '=', move_browse.product_id.id), ('bom_id', '=', production_browse.bom_id.id), ('subproduct_type', '=', 'fixed')], context=context)
```

Comment:
I think, no need to search twice.

I propose to use search_read in line#120 to read fields:  subproduct_type, product_qty
and return qty base on subproduct_type.